### ISSUE TITLE
CLI: Build for Apple M1

### ISSUE THREAD

**Issue by marcstreeter** [argoproj/argo-workflows#5745]
I notice that there is a Darwin AMD64 and was wondering if building it for the M1 would be supported?

**Comment by alexec** [comment]
Not yet. Don't even know if it works. Would you like to try it?

**Comment by alexec** [comment]
Looks like we'd need to upgrade to Go 1.16:

https://www.zdnet.com/article/go-programming-language-just-added-native-support-for-apples-mac-m1-arm-chip/

**Comment by alexec** [comment]
#5563

**Comment by marcstreeter** [comment]
> Not yet. Don't even know if it works. Would you like to try it?

Definitely! In the meantime I'll see if I can just get along with using Rosetta.

**Comment by terrytangyuan** [comment]
@alexec May I ask what's the reason for closing https://github.com/argoproj/argo-workflows/pull/5721 that could fix #5563?

**Comment by alexec** [comment]
I closed #5721 because it was intended as PoC, I wanted to see what benefits 1.16 might have. I didn't feel that I was getting much out of it.

**Comment by yriveiro** [comment]
I believe this one can be closed as the current argo cli is built using `darwin/arm64` architecture already

```bash
argo: v3.2.7+bdc14b0.dirty
  BuildDate: 2022-01-28T04:34:09Z
  GitCommit: bdc14b0d2e50b6d3b30f7b6ffbfd8ceae3214413
  GitTreeState: dirty
  GitTag: v3.2.7
  GoVersion: go1.17.6
  Compiler: gc
  Platform: darwin/arm64
```
